components/user_settings/sidebar/user_settings_sidebar.jsx

```diff
@@ -58,32 +82,79 @@ export default class UserSettingsSidebar extends React.Component {
     }
 
     getStateFromStores = () => {
-        const {closeUnusedDirectMessages, displayUnreadSection} = this.props;
+        const {
+            closeUnusedDirectMessages,
+            sidebarPreference: {
+                grouping,
+                sorting,
+                unreadsAtTop,
+                favoriteAtTop,
+            },
+        } = this.props;
+
         return {
             settings: {
                 close_unused_direct_messages: closeUnusedDirectMessages,
-                show_unread_section: displayUnreadSection,
+                grouping,
+                unreadsAtTop,
+                favoriteAtTop,
+                sorting,
             },
             isSaving: false,
         };
     };
 
+    trackSettingChangeIfNecessary = (setting) => {
+        if (this.state.settings[setting] !== this.props.sidebarPreference[setting]) {
+            trackEvent('settings', 'user_settings_update', {field: 'sidebar.' + setting, value: this.state.settings[setting]});
+        }
+    }
+
     updateSetting = (setting, value) => {
         const settings = this.state.settings;
         settings[setting] = value;
+
         this.setState(settings);
     };
 
     handleSubmit = (setting) => {
         const {actions, user} = this.props;
         const preferences = [];
 
-        preferences.push({
-            user_id: user.id,
-            category: Constants.Preferences.CATEGORY_SIDEBAR_SETTINGS,
-            name: setting,
-            value: this.state.settings[setting],
-        });
+        if (setting === 'channel_grouping' || setting === 'channel_sorting') {
+            const updatedSidebarSettings = {
+                grouping: this.state.settings.grouping,
+                unreads_at_top: this.state.settings.unreadsAtTop,
+                favorite_at_top: this.state.settings.favoriteAtTop,
+                sorting: this.state.settings.sorting,
+            };
+
+            if (updatedSidebarSettings.grouping === 'by_type') {
+                // Note that this isn't tracked by the analytics below because it's not in this.state
+                updatedSidebarSettings.sorting = 'alpha';
```

Comment:
I think the whole point to have an experimental feature is to track how many users do use it in order for us to decide if we keep it or not, can we make sure this is added to the analytics?

Reply:
@hmhealey can you explain this note a bit more

Reply:
I can actually remove that if statement entirely. I think the reason that this was there was because there used to be two separate settings entries for grouping and sorting, and since the "by type" grouping ignores the sorting preference, it was changing that so that it shows correctly in the UI